### PR TITLE
Fix sessionId handling to allow authentication with sessionId

### PR DESCRIPTION
**Pull request by chrbayer**
- The sessionId was used manually for some ParseUser communication, but has to be used with all communication with the parse server to be able to access protected resources. It took me quite some time to find me the reason why I was not able to access my classes after I activating CLP for my classes...

 - Some small code clean-ups regarding storage.

### CODE REVIEW

**Comment by phillwiggins** [comment]
Hey, can you check my comments please.

Just a few queries.

Cheers

**Comment by chrbayer** [comment]
Hi, of course, but where can I find your comments? I'm not using github that often :-)

**Comment by phillwiggins** [comment]
Ah okay, they are on GitHub.

There seems to be some confusion on when your removing token from one
method and when your replacing it elsewhere.

**Comment by chrbayer** [comment]
In the previous version, the sessionId was manually inserted in the client request in three methods of ParseUser: getCurrentUserFromServer(), save() and destroy(). I removed this, instead the the sessionId is now always send in any transaction to the server if available, I included it in ParseHTTPClient.send(). In a real parse server application I think it is important to restrict the access to any used classes on the server, if not there would be no reason to signUp or login at all. And if the access is restricted, you need to authenticate each access with your sessionId.

**Comment by phillwiggins** [comment]
I see. So the user token is only sent for user based requests?

**Comment by chrbayer** [comment]
No, for all requests. Is not every request somehow user based after login?

**Comment by phillwiggins** [comment]
Okay so after further reading, I think this makes sense when used with ACL. As we don't always want to send a token, please can you change your code so that the user has the option of sending a token. @chrbayer it's good work, but not quite there. This doesn't fit all user requirements. Maybe set a bool in ParseCoreData for now that can determine whether to send token with all requests?

File: lib/parse_server_sdk.dart [hunk 1]
```diff
@@ -89,8 +89,6 @@ class Parse {
         sessionId: sessionId,
         securityContext: securityContext);
 
-    ParseCoreData().initStorage();
```
[the remark this answers is not in the document]
Comment (chrbayer) [reply lib/parse_server_sdk.dart]: The init functionality was and still is in getStore(), It is initialized at the first usage:

Future<SharedPreferences> getStore() async {
  return storage ?? (storage = await SharedPreferences.getInstance());
}

File: lib/src/objects/parse_user.dart [hunk 2]
```diff
@@ -63,20 +63,16 @@ class ParseUser extends ParseObject implements ParseCloneable {
   /// fromServer can be called and an updated version of the [User] object will be
   /// returned
   static Future<ParseResponse> getCurrentUserFromServer(
-      {String token, bool debug, ParseHTTPClient client}) async {
```
[the remark this answers is not in the document]
Comment (chrbayer) [reply lib/src/objects/parse_user.dart]: Oh, Ok, I was not aware it was recently added, I will look for a way to use a different token in this function.

**Comment by chrbayer** [comment]
Even if the the sessionId is not needed for ACL or CLP, what harm does it make sending it? Nevertheless I can make sending the sessionId optional for not user objects.

**Comment by phillwiggins** [comment]
I see.

I'm not sure if there are many side affects of having the user token
present when not required, but looking through the docs, it still doesn't
look like best practice.

On this basis, please do make it an optional. I think somebody is looking
into implementing ACL features as we speak and I am hoping they will make
the call on this.

I really do appreciate your efforts.

**Comment by phillwiggins** [comment]
I've merged this but will need to do some work.

Your still forcing tokens when it's not necessary. The only token that users are going to be sending is their current user token, so checking if a token is present in the headers and if not using the token from ParseCoreData is redundant. It's going to be the same token either way.

**Comment by chrbayer** [comment]
Yeah, I am not ready and still working on making sending the token optional, I was interrupted this afternoon :-) 

Checking if a header is present is not completely redundant, because in the getCurrentUserFromServer() function I do insert a user given not necessarily the same SessionId header.

I inform you as soon as my work is done.

Thanks!

**Pull request merged**